CRM/Contribute/BAO/Contribution.php

```diff
@@ -2169,24 +2166,14 @@ public static function transitionComponents($params) {
     if ($contributionStatus === 'Completed') {
 
       // only pending contribution related object processed.
-      if ($previousContriStatusId &&
-        !in_array($previousStatus, [
+      if (!in_array($previousStatus, [
```

Comment:
previous status will be a status will be FALSE if $previousContriStatusId  is empty. FALSE is NOT in ['Pending', 'Partially Paid'] so we don't need to check both